src/Microsoft.DotNet.Build.Tasks/PackageFiles/RunnerTemplate.Unix.txt

```diff
@@ -50,10 +50,10 @@ if [ "$(uname -s)" == "Darwin" ]; then
   # On OS X, we will enable core dump generation only if there are no core 
   # files already in /cores/ at this point. This is being done to prevent
   # inadvertently flooding the CI machines with dumps.
-  if [ -d "/cores" ]; then
-    if [ ! "$(ls -A /cores)" ]; then
-      ulimit -c unlimited
-    fi
+  if [ ! -d "/cores" ]; then
+    ulimit -c unlimited
+  elif [ ! "$(ls -A /cores)" ]; then
```

Comment:
The or operator checks the second statement even if the first statement is true so I added else if